ui/StatusQ/include/StatusQ/writableproxymodel.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <QAbstractProxyModel>
+
+// WritableProxyModel is a QAbstractProxyModel that allows you to modify the data without modifying the source model.
+// It is useful for implementing a "dirty" state for a model, where you can modify the data and then commit the changes
+// to the source model.
+
+// Supported features (reimplemented):
+// - setData
+// - setItemData
+// - removeRows
+// - insertRows
+// - moveRows (TODO)
+// - toVariantMap
+// - to be continued...
+
+// Limitations:
+// - only 1 column (no grid models)
+// - no parent index (no tree models)
+
+class WritableProxyModelPrivate;
+
+class WritableProxyModel : public QAbstractProxyModel
+{
+    Q_OBJECT
+
+    Q_PROPERTY(bool dirty READ dirty NOTIFY dirtyChanged)
+
+public:
+    explicit WritableProxyModel(QObject* parent = nullptr);
+    ~WritableProxyModel();
+
+    Q_INVOKABLE QVariantMap toVariantMap() const;
```

Comment:
This is up for discussion. Its purpose is to flush the entire model in QVariantMap. This can be used to synchronise with backend once the editing is finished.
It currently returns the entire model in a format similar to what [itemData](https://doc.qt.io/qt-5/qabstractproxymodel.html#itemData) is implementing. Processing this in QML could be tricky, but I think it should not be processed in any way in qml.

If you think it would be nice to have a qml friendly format we could have an implementation with a QQmlPropertyMap using role names for each row.

Reply:
I think this is fine; SFPM has these for reference:
```cpp
Q_INVOKABLE QVariantMap get(int row) const;
Q_INVOKABLE QVariant get(int row, const QString& roleName) const;
```